src/ray/gcs/format/gcs.fbs

```diff
@@ -89,33 +91,38 @@ table CustomSerializerData {
 table ConfigTableData {
 }
 
+table RayResource {
+  // The type of the resource.
+  resource_name: string;
+  // The total capacity of this resource type.
+  resource_capacity: double;
+}
+
 table ClientTableData {
   // The client ID of the client that the message is about.
   client_id: string;
   // The IP address of the client's node manager.
   node_manager_address: string;
   // The port at which the client's node manager is listening for TCP
   // connections from other node managers.
-  local_scheduler_port: int;
+  node_manager_port: int;
   // The port at which the client's object manager is listening for TCP
   // connections from other object managers.
   object_manager_port: int;
   // True if the message is about the addition of a client and false if it is
   // about the deletion of a client.
   is_insertion: bool;
+  resources_total_label: [string];
+  resources_total_capacity: [double];
 }
 
-table Resource {
-  // The type of the resource.
-  resource_name: string;
-  // The total capacity of this resource type.
-  resource_capacity: double;
-}
-
-table NodeManagerHeartbeat {
-  // The available resources on this node manager. This information may be
-  // stale.
-  resources_available: [Resource];
-  // The total resources on this node manager.
-  resources_total: [Resource];
+table HeartbeatTableData {
```

Comment:
@atumanov, we recently merged some commits into master that should allow you to use nested flatbuffers now. Once we merge those into this branch, can you see if using `RayResource` directly works for you now?